### PR TITLE
feat(migrations): Add migration for onpremise compatibility

### PR DESCRIPTION
**Pull request by lynnagara**
This migration provides compatibility for any onpremise user that
is migrating from a pre- migration system version of Snuba. We need
to sync up all of the columns that have changed in the sentry_local
individually to ensure all of these get the correct state going
forward.

We would need to provide something similar for the other two tables
(transactions and errors) that have had their schema changed since
they were first introduced.

### CODE REVIEW

File: snuba/migrations/snuba_migrations/events/0002_onpremise_compatibility.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    This is a one-off migration to support on premise users who are upgrading from
+    any older version of Snuba that used the old migration system. Since their sentry_local
+    table might be previously created with slightly different columns, this migration
+    should bring them back in sync by by adding and removing the relevant columns
```
Comment (fpacifici) [inline snuba/migrations/snuba_migrations/events/0002_onpremise_compatibility.py:21]: remove one `by` please

**Review by fpacifici — COMMENTED** [review]
One clarification: is this for users that are on ancient version of snuba, that is older than the migrations we added in the old system and who want to directly jump to a version that does not use the old migration system anymore ?

**Comment by lynnagara** [comment]
@fpacifici Yeah - it's to support moving from any version to any version of Snuba in on premise. The alternative to this is building something that forces an upgrade to some intermediate Snuba version first before moving onto the latest one. We don't have such a system in place now to either enforce this or make the upgrade process not painful, so I think it's better to just run this migration for everyone instead. It's harmless if they're up to date.

**Review by fpacifici — APPROVED** [review]
ok, understood, thanks

**Pull request merged**